### ISSUE TITLE
[ docs ] add a note to builtin components docs explaining that + means required

### ISSUE THREAD

**Issue by flynntsang** [eidolon-ai/eidolon#803]
**Problem:** In the tables at the top of builtins documentation, it's not obvious why some properties are preceded with + and other by -. 

**To do:** 
- Add a note to the pages indicating that + indicates required values
- Or
  - remove the minus sign - for optional fields
  - change the plus sign + to an asterisk * which is more commonly understood to mean required.

[Example:](https://www.eidolonai.com/docs/components/logicunit/apilogicunit)

![Image](https://github.com/user-attachments/assets/e8cc29c6-e813-4b02-9a92-da7f500afdf5)

**Comment by LukeLalor** [comment]
This is generated by "json schema for humans" in docbuilder.py

Note, we could also consider
1. making our own markdown template
2. using the html generator instead

**Comment by LukeLalor** [comment]
throwing the needs more info tag on to decide how deeply we want to refactor our auto docs since right now they kinda suck

**Comment by flynntsang** [comment]
Update: let's not gold-plate the request if we don't need to. If it's simple enough to append a note on each generated page that ` [ + ] ` indicates required we're good for now. The objective at the moment is to make the page more readily understandable to the person reading the page.

**Comment by LukeLalor** [comment]
yes, we can add that to the template above the table without complexity
